### PR TITLE
Stop cloud dl

### PR DESCRIPTION
**Pull request by macpijan**
The goal of this MR:
- replace: https://github.com/Dasharo/open-source-firmware-validation/pull/485 
- enable ESP scanning tests on more boards (PiKVM is not realllly a requirement here)
- enable Secure Boot tests on more boards (on non-PiKVM we would need to insert stick with [this image](https://github.com/Dasharo/osfv-test-data/tree/main/secure-boot) - which is now integrated into a single image, as opposed to mulltiple images before)

What should be tested:
- [x] full regression on QEMU
  - some outstanding errors in area of disks handling - won't be handled here: https://github.com/Dasharo/open-source-firmware-validation/issues/680
- [x] ESP scanning on PiKVM
- [x] ESP scanning on non-PiKVM
- [x] bios lock/earlydma/TPM/vboot on PiKVM
- [x] SB on PiKVM
- [x] DTS on non-PiKVM
- [x] SB on non-PiKVM
- [x] DTS on PiKVM
- [x] bios lock/earlydma/TPM/vboot on non-PiKVM
- [x] capsule update - at least logo persistence test

Suggested order of merging:
- [x] https://github.com/Dasharo/open-source-firmware-validation/pull/661
- [x] https://github.com/Dasharo/open-source-firmware-validation/pull/659
- [x] https://github.com/Dasharo/open-source-firmware-validation/pull/665
- [ ] This MR (after rebase)

### CODE REVIEW

**Comment by filipleple** [comment]
@macpijan  Getting instant fails here due to this issue: https://github.com/Dasharo/open-source-firmware-validation/issues/628 so I've done 
```
@@ -36,7 +36,7 @@ Get Hostname Ip
     # TODO: We do not necessarily need Internet to be reachable for the internal
     # addresses to be assigned. But if it is, the local IPs are definitely
     # already in place.
-    Wait Until Keyword Succeeds    5x    1s
+    Wait Until Keyword Succeeds    5x    10s
     ...    Check Internet Connection On Linux
     ${out_hostname}=    Execute Command In Terminal    hostname -I
     Should Not Contain    ${out_hostname}    link is not ready
     
```

Now, despite doing 
```
git submodule update --init --checkout
virtualenv -p $(which python3) robot-venv
source robot-venv/bin/activate
pip install -U -r requirements.txt
```
after checking out the branch, I get

```
ESP001.001 ESP Scan with OS-specific .efi files added :: This test... | FAIL |
Parent suite setup failed:
There are no source files matching '/home/flewinski/workspace/vp6650-test/open-source-firmware-validation/osfv-test-data/esp-scanning/q
emu-disk.img'
```

So I'm guessing `osfv-test-data` is not properly registered as submodule/cloned by tests which need it? After cloning it manually I got

```
==============================================================================
Esp-Scanning
==============================================================================
ESP001.001 ESP Scan with OS-specific .efi files added :: This test... | FAIL |
[ Ubuntu (on SSDPR-CL100-240-G2) | Windows Boot Manager | Windows Boot Manager (on SSDPR-CL100-240-G2) | SSDPR-CL100-240-G2 | Network Boot and Utilities | UEFI Shell | Setup | USB SanDisk 3.2Gen1 | Dasharo Tools Suite (on USB SanDisk 3.2Gen1) ] does not contain match for pattern 'Suse Boot Manager (on*'.
------------------------------------------------------------------------------
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... ..Mounting USB Disk Image at runtime is not supported on this platform.
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... | SKIP |
Image not marked as required, skipping test case.
------------------------------------------------------------------------------
ESP004.001 ESP Scan does not create duplicate entries :: This test... | PASS |
------------------------------------------------------------------------------
ESP005.001 ESP Scan detects Dasharo Tools Suite :: This test aims ... ..Mounting USB Disk Image at runtime is not supported on this platform.
Image marked as required. Make sure that USB drive with image: /home/flewinski/workspace/vp6650-test/open-source-firmware-validation/osfv-test-data/dts/dts-base-image-v2.1.3.wic is already prepared and connected to the DUT.
ESP005.001 ESP Scan detects Dasharo Tools Suite :: This test aims ... | PASS |
------------------------------------------------------------------------------
ESP006.001 ESP Scan does not find non-block boot devices :: This t... | PASS |
------------------------------------------------------------------------------
ESP002.001 ESP Scan after deleting additional .efi files :: This t... | PASS |
------------------------------------------------------------------------------
Since the asset has been checkout manually by you prior running this script, it will NOT be checked in automatically. Please return the device when work is finished.
Esp-Scanning                                                          | FAIL |
6 tests, 4 passed, 1 failed, 1 skipped
==============================================================================
```

Some tests pass, but after looking at the log more closely, seems like the test doesn't manage to add any OSes to boot menu:
```
Arguments: [ ['Ubuntu (on SSDPR-CL100-240-G2)',
 'Windows Boot Manager',
 'Windows Boot Manager (on SSDPR-CL100-240-G2)',
 'SSDPR-CL100-240-G2',
 'Network Boot and Utilities',
 'UEFI Shell',
 'Setup',
 'USB SanDisk 3.2Gen1',
 'Dasharo Tools Suite (on USB SanDisk 3.2Gen1)'] | 'Suse Boot Manager (on*' ]	
13:16:45.365	FAIL	[ Ubuntu (on SSDPR-CL100-240-G2) | Windows Boot Manager | Windows Boot Manager (on SSDPR-CL100-240-G2) | SSDPR-CL100-240-G2 | Network Boot and Utilities | UEFI Shell | Setup | USB SanDisk 3.2Gen1 | Dasharo Tools Suite (on USB SanDisk 3.2Gen1) ] does not contain match for pattern 'Suse Boot Manager (on*'.
```
So the non-PiKVM approach might require some extra work

**Comment by macpijan** [comment]
There is another serious problem with this MR:

```
Git LFS has been disabled on the organization Dasharo because you’ve exceeded your data plan by at least 150%. Please purchase additional data packs to cover your bandwidth and storage usage:

  https://github.com/organizations/Dasharo/billing/data/upgrade

Current usage as of 16 Jan 2025 12:27AM UTC:

  Bandwidth: 0.57 GB / 1 GB (56%)
  Storage: 1.6 GB / 1 GB (160%)
```

So gettint these test files from github is not a good idea :disappointed:

**Comment by macpijan** [comment]
I have resullts on non-piKVM yesterday:


Odroid ESP scanning:

```
Watch "logs/odroid-h4-plus/2025_01_16_00_09_38/dasharo-compatibility/esp-scanning.robot_debug.log" to monitor the progress of the test
==============================================================================
Esp-Scanning                                                                  
==============================================================================
ESP001.001 ESP Scan with OS-specific .efi files added :: This test... | PASS |
------------------------------------------------------------------------------
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... ..Mounting USB Disk Image at runtime is not supported on this platform.
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... | SKIP |
Image not marked as required, skipping test case.
------------------------------------------------------------------------------
ESP004.001 ESP Scan does not create duplicate entries :: This test... | PASS |
------------------------------------------------------------------------------
ESP005.001 ESP Scan detects Dasharo Tools Suite :: This test aims ... ..Mounting USB Disk Image at runtime is not supported on this platform.
Image marked as required. Make sure that USB drive with image: /home/macpijan/projects/github/dasharo/open-source-firmware-validation/osfv-test-data/dts/dts-base-image-v2.1.3.wic is alredy prepared and connected to the DUT.
ESP005.001 ESP Scan detects Dasharo Tools Suite :: This test aims ... | PASS |
------------------------------------------------------------------------------
ESP006.001 ESP Scan does not find non-block boot devices :: This t... | PASS |
------------------------------------------------------------------------------
ESP002.001 ESP Scan after deleting additional .efi files :: This t... | PASS |
------------------------------------------------------------------------------
Since the asset has been checkout manually by you prior running this script, it will NOT be checked in automatically. Please return the device when work is finished.
Esp-Scanning                                                          | PASS |
6 tests, 5 passed, 0 failed, 1 skipped
```

So the logic is probably fine, but the GH LFS is a big miss.

**Comment by filipleple** [comment]
Hmm, I've got the same result again, using the repo mirror from cloud instead of gh lvs. Not sure what makes the VP6650 a corner case.

```
==============================================================================
ESP001.001 ESP Scan with OS-specific .efi files added :: This test... | FAIL |
[ Ubuntu (on SSDPR-CL100-240-G2) | Windows Boot Manager | Windows Boot Manager (on SSDPR-CL100-240-G2) | SSDPR-CL100-240-G2 | Network Boot and Utilities | UEFI Shell | Setup ] does not contain match for pattern 'Suse Boot Manager (on*'.
------------------------------------------------------------------------------
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... ..Mounting USB Disk Image at runtime is not supported on this platform.
ESP003.001 ESP Scan ignores OSes on removable media :: This test a... | SKIP |
Image not marked as required, skipping test case.
------------------------------------------------------------------------------
ESP004.001 ESP Scan does not create duplicate entries :: This test... | PASS |
------------------------------------------------------------------------------
```


[Here's the full log](https://cloud.3mdeb.com/index.php/s/bAn5CBNG5HJfDbW)

**Review by filipleple — APPROVED** [review]

**Comment by macpijan** [comment]
@filipleple Do you have some logs/results to be shared here on top of approval?

**Comment by filipleple** [comment]
@macpijan logs can be previewed [here](https://cloud.3mdeb.com/index.php/s/m7R7jXsLN2XonNE). Full logs for secure-boot and dts are missing for the moment, I need to dump them again in the background

EDIT: added the missing logs

**Comment by filipleple** [comment]
@macpijan  ESP scanning fixed, log uploaded to the same directory as previous ones

**Comment by macpijan** [comment]
@filipleple FYI, we have a script like:

```
./scripts/refactoring-state.sh 
Keywords that should not be used, but are still used:
./platform-configs/raptor-cs_talos2.robot:253:        Sonoff Power On
./dasharo-security/smm-bios-write-protection.robot:50:    Get Flashrom From Cloud
./dasharo-security/smm-bios-write-protection.robot:74:    Get Flashrom From Cloud
./lib/dcu.robot:55:    Get Flashrom From Cloud
./lib/options/options-lib_dcu.robot:130:    Get Flashrom From Cloud
./lib/power-after-fail-lib.robot:12:    ...    via a RTE relay or a sonoff.
./util/basic-platform-setup.robot:117:    Get Cbmem From Cloud
Global variables that should not be used, but are still used:
```

**Comment by macpijan** [comment]
I modified SMM, so confirming it still works:

```
FW_FILE=protectli_vp66xx_v0.9.1.rom DEVICE_IP=192.168.10.132 RTE_IP=192.168.10.110 CONFIG=protectli-vp6670 ./scripts/run.sh dasharo-security/smm-bios-write-protection.robot
Logs will be saved at logs/protectli-vp6670/2025_03_14_12_28_02
Watch "logs/protectli-vp6670/2025_03_14_12_28_02/dasharo-security/smm-bios-write-protection.robot_debug.log" to monitor the progress of the test
==============================================================================
Smm-Bios-Write-Protection                                                     
==============================================================================
SMM001.001 SMM BIOS write protection enabling (Ubuntu) :: SMM BIOS... | PASS |
------------------------------------------------------------------------------
SMM002.001 SMM BIOS write protection disabling (Ubuntu) :: SMM BIO... | PASS |
------------------------------------------------------------------------------
The asset has been checked out automatically. It is now automatically checked in as well. If you want to keep using it, reserve it manually first.
Smm-Bios-Write-Protection                                             | PASS |
2 tests, 2 passed, 0 failed
==============================================================================
```

**Comment by macpijan** [comment]
DCU works as well:

```
Dcu                                                                           
==============================================================================
DCU001.001 Change the UUID :: This test case verifies that the UUI... | PASS |
------------------------------------------------------------------------------
DCU002.001 Change the serial number :: This test case verifies tha... | PASS |
```
